File: api/src/endpoint/user/list_user_organizations.rs
use serde::{Deserialize, Serialize};

use api_derive::ApiRequest;

use crate::{endpoint::Endpoint, resource::OrganizationResource};

pub struct ListUserOrganizations;

impl Endpoint for ListUserOrganizations {
    const PATH: &'static str = "/user/{user_name}/organizations";
    const METHOD: http::Method = http::Method::GET;

    type Request = ListUserOrganizationsRequest;
    type Response = ListUserOrganizationsResponse;
}

#[derive(ApiRequest, Debug, Serialize, Deserialize)]
pub struct ListUserOrganizationsRequest {}

pub type ListUserOrganizationsResponse = Vec<OrganizationResource>;

File: gitdot-api-ts/src/endpoint/user/list-user-organizations.ts
import { z } from "zod";
import { OrganizationResource } from "../../resource";
import type { Endpoint } from "../endpoint";

export const ListUserOrganizationsRequest = z.object({});
export type ListUserOrganizationsRequest = z.infer<
  typeof ListUserOrganizationsRequest
>;

export const ListUserOrganizationsResponse = z.array(OrganizationResource);
export type ListUserOrganizationsResponse = z.infer<
  typeof ListUserOrganizationsResponse
>;

export const ListUserOrganizations = {
  path: "/user/{user_name}/organizations",
  method: "GET",
  request: ListUserOrganizationsRequest,
  response: ListUserOrganizationsResponse,
} satisfies Endpoint;
export type ListUserOrganizations = typeof ListUserOrganizations;
